### ISSUE TITLE
Fdroid repository initialization

### ISSUE THREAD

**Issue by andrealinux1** [NoMore201/playmaker#2]
First of all __thank you__ for this awesome project, I used gplayweb in the past but it stopped working at some point and I manually fetched and installed the _apk_ files of the various applications since then. This project is the perfect fit for my current Android setup.

I deployed __playmaker__ on a server taking advantage of the pre-built image on dockerhub and all worked well out of the box, so I proceeded to replicate the same setup that I had with gplayweb in the previous installation. 
In particular I use __nginx__ to _serve_ the fdroid repository in order to reach it for example from the fdroid client on my phone.

For doing this I use a combination of two containers, that I deploy with __docker-compose__, with the following configuration:

```docker
docker-compose.yml
andrea@kimsufi:~/docker/playmaker-hub$ cat docker-compose.yml 
version: '3'
services:
  playmaker:
    image: nomore201/playmaker
    restart: always
    expose:
      - "5000"
    volumes:
      - /home/docker/playmaker:/data/fdroid
    environment:
     ...
    network_mode: "bridge"    


  fdroid-nginx:
    image: nginx      
    restart: always
    expose:
      - "80"
    volumes:
      - /home/docker/playmaker/repo:/usr/share/nginx/html:ro
    environment:
      ...
    network_mode: "bridge"

```

For readability reasons I omitted some parts that are not significant for this discussion (the environment variables used for instructing the __nginx-proxy__ container on which third-level domain expose the container).

With this setup I incurred in a problem: since we instruct the __fdroid-nginx__ container to mount the `/home/docker/playmaker/repo` folder in a location that will be served by nginx, the docker daemon takes care of creating the folder in case it doesn't exists. 
In this way the __playmaker__ container when executing the startup routine doesn't initialize the fdroid repository since the repo folder is already present.
The check is done in the following place:
https://github.com/NoMore201/playmaker/blob/2eaba47499b60adff4c42d0da108ba4d7bdf9aa9/playmaker/service.py#L45
This causes the fdroid repository not being initialized even at the first startup.

Obviously this is not a problem of playmaker itself, in fact if we create as first thing the playmaker container and only after start the nginx container that expose the repository, all works correctly. Anyway I decided to report this since I think that the majority of the users could do a setup similar to the one I showed, incurring in the problem (I know for sure that @n1zzo uses a similar setup).

Maybe it would be nice to mention the fact that the initialization of the repository is not executed if the 'repo' folder is present, or check for multiple signs that the initialization has occurred (presence of the `config.py` or `keystore.jks` files in addition to the `repo` folder for example), or even use an helper function of fdroid to check this (I'm speculating here, I will check for this in the fdroid documentation).

Let me know which solution you find more appropriate, and I will gladly make a __PR__ for that.

**Comment by NoMore201** [comment]
Yes, for sure checking for the presence of fdroid related files would be a better approach. I did some testing with the [fdroidserver tool](https://gitlab.com/fdroid/fdroidserver) and it shows the following behaviour:

In an empty folder, containing only a *repo* subdirectory, it will throw an error like this:

```
$ fdroid stats
CRITICAL: missing config file - is this a repo directory?
```

The main file seems to be `config.py`. If the `keystore.jks` is missing, the fdroid tool will consider the repo initializated, and it can be created again using `fdroid update --create-key`. Any other folder like `archive`, `metadata`, `repo` will be reinitialized with each update call. So i guess the correct steps are the following:

1 Check for a `config.py`
2 If not exists call `fdroid init`
3 Call `fdroid update --create-key`, it will re initialize all the missing data.

**Comment by NoMore201** [comment]
04b66b3 should fix this, right now I will test it and then upload the new version on Docker hub.

**Comment by NoMore201** [comment]
Updated Docker image, let me know the problem is solved

**Comment by andrealinux1** [comment]
Sorry for the delay, I did not see the notification for this discussion (I have a separate e-mail address for github notifications, and obviously I forgot to check it).
I updated the image from the Docker Hub and reinitialized the container from scratch. Now the initialization works flawlessly as one would expect. I also noticed a lot of awesome improvements in the UI and also in *under the hood* details.
Again a lot of thanks for making this project happen, you have improved **immeasurably** my daily smartphone experience.
